pkg/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,239 @@
+package networkpolicy
+
+import (
+	"cmp"
+	"fmt"
+	"net"
+	"slices"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	npav1alpha1 "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+)
+
+func (c *Controller) adminNetworkPolicyAction(p packet) npav1alpha1.AdminNetworkPolicyRuleAction {
+	srcIP := p.srcIP
+	srcPod := c.getPodAssignedToIP(srcIP.String())
+	srcPort := p.srcPort
+	dstIP := p.dstIP
+	dstPod := c.getPodAssignedToIP(dstIP.String())
+	dstPort := p.dstPort
+	protocol := p.proto
+	srcPodAdminNetworkPolices := c.getAdminNetworkPoliciesForPod(srcPod)
+	dstPodAdminNetworkPolices := c.getAdminNetworkPoliciesForPod(dstPod)
+	msg := fmt.Sprintf("checking packet %s:", p.String())
+	if srcPod != nil {
+		msg += fmt.Sprintf(" SrcPod (%s/%s): %d AdminNetworkPolicy", srcPod.Name, srcPod.Namespace, len(srcPodAdminNetworkPolices))
+	}
+	if dstPod != nil {
+		msg += fmt.Sprintf(" DstPod (%s/%s): %d AdminNetworkPolicy", dstPod.Name, dstPod.Namespace, len(dstPodAdminNetworkPolices))
+	}
+	klog.V(2).Infof("%s", msg)
+
+	// For a connection from a source pod to a destination pod to be allowed,
+	// both the egress policy on the source pod and the ingress policy on the
+	// destination pod need to allow the connection.
```

Comment:
I think it is a bit more complicated than that. A condition for a successful connection (ignoring baseline ANP) is something like
`egress Allow && ingress Allow` =
`(egress ANP Allow || egress ANP skip && egress NP Allow) && (ingress ANP Allow || ingress ANP Skip && ingress NP Allow)`
An example match may be:
egress ANP Skip, egress NP Allow => egress allow
ingress ANP Allow, ingress NP Drop => ingress allow
With current logic it will be ANP verdict => Skip, NP verdict =>Drop, result => Drop.

So with admin network policies we may need to get verdicts for ingress and egress separately and then evaluate an expression like this ^

Reply:
I see this simpler as 2 checks, 

check 1: ANP it can be "not apply", allowed, skipped or denied

check 2. Network policies, only comes here if 1 does not apply or is skipped, no?

Reply:
I think some unit tests for anp will help us understand this case better

Reply:
yep, yep

Reply:
almost there, half way with the tests